### PR TITLE
CI and python script should include java-security-sample-ias

### PR DESCRIPTION
**Pull request by nenaraab**
- the service and service key was not successfully deleted

### CODE REVIEW

File: samples/deploy_and_test.py [hunk 1]
```diff
@@ -789,27 +805,27 @@ def __str__(self):
...
-        subprocess.run(
-            ['cf', 'create-service', 'xsuaa', 'application', self.xsuaa_service_name, '-c', 'xs-security.json'],
-            cwd=self.working_dir, stdout=cf_logs, check=True)
+        if self.xsuaa_service_name is not None:
+            logging.info("Creating Xsuaa service '{}'".format(self.xsuaa_service_name))
+            subprocess.run(
+                ['cf', 'create-service', 'xsuaa', 'application', self.xsuaa_service_name, '-c', 'xs-security.json'],
+                cwd=self.working_dir, stdout=cf_logs, check=True)
         if self.identity_service_name is not None:
-            logging.info("Creating IAS service '{}'".format(self.identity_service_name))
```
Comment (liga-oz) [inline samples/deploy_and_test.py:810]: this logging output should be added in `__create_ias_service()` method

File: samples/deploy_and_test.py [hunk 2]
```diff
@@ -819,14 +835,16 @@ def deploy(self):
     def delete(self):
         logging.info("Deleting '{}'".format(self.name))
         subprocess.run(['cf', 'delete', '-f', '-r', self.name], stdout=cf_logs)
+        subprocess.run(['cf', 'delete-orphaned-routes', '-f'], stdout=cf_logs)
         if self.app_router_name is not None:
             logging.info("Deleting '{}' app router".format(self.app_router_name))
             subprocess.run(['cf', 'delete', '-f', '-r', self.app_router_name], stdout=cf_logs)
         logging.info("Deleting '{}' Xsuaa service".format(self.xsuaa_service_name))
```
Comment (liga-oz) [inline samples/deploy_and_test.py:842]: this logging info should then go also in the if statement L844

File: samples/deploy_and_test.py [hunk 3]
```diff
@@ -498,7 +512,7 @@ def __execute(self, req):
 
 class IasAccess:
     def __init__(self, ias_name):
-        self.ias_service_key_name = "ias-access-key"
+        self.ias_service_key_name = ias_name + '-key'
```
Comment (liga-oz) [inline samples/deploy_and_test.py:515]: really good one 🥇

**Review by liga-oz — CHANGES_REQUESTED** [review]
just few minor changes for logging output, other then that very good :)

File: samples/deploy_and_test.py (hunk 2 above)
Comment (nenaraab) [reply samples/deploy_and_test.py:849]: not sure, delete-orphaned-routes deletes routes from the application itself...

File: samples/deploy_and_test.py (hunk 1 above)
Comment (nenaraab) [reply samples/deploy_and_test.py:810]: yep, done :-)

File: samples/deploy_and_test.py [hunk 4]
```diff
@@ -584,17 +586,12 @@ def fetch_ias_token(self, user):
...
 
     def delete(self):
-        IasAccess.delete(self.ias_service_name)
-
-    @staticmethod
-    def delete(ias_service_name):
-        ias_service_key_name = ias_service_name + '-key'
         logging.info("Deleting service key '{}' for '{}' IAS service"
-                     .format(ias_service_key_name, ias_service_name))
+                     .format(self.ias_service_key_name, self.ias_service_name))
```
Comment (nenaraab) [inline samples/deploy_and_test.py]: Hi @liga-oz is it a good practice to use self in a static method? instead we want to use parameter and local variable...

**Review by nenaraab — COMMENTED** [review]
hi @liga-oz thanks for the nice fix!

**Pull request merged**